test/dynamo/test_functions.py

```diff
@@ -450,6 +450,31 @@ def test_callable_builtin(x):
         else:
             return x - 1
 
+    def test_callable_class(self):
+        class CallableClass:
+            def __call__():
+                pass
+
+        class NotCallableClass:
+            pass
+
+        @torch.compile(backend="eager", fullgraph=True)
+        def fn(x, arg):
+            if callable(arg):
+                return x
+            return x + 1
+
+        input = torch.randn(4)
+
+        self.assertEqual(fn(input, NotCallableClass()), input)
+        self.assertEqual(fn(input, CallableClass()), input)
+
+        # passing tensor and scalars
+        self.assertEqual(fn(input, 1), input + 1)
+        self.assertEqual(fn(input, 1.1), input + 1)
```

Comment:
Would suggest to use tensor here. There is some piece of code somewhere in Dynamo which check is this function is even worth of compilation, and it checks that either the inputs or the function itself should have something torch.Tensor related.

Reply:
input is a tensor. I am not sure i understood the comment